### PR TITLE
VACMS-3303: Update composer to v2 and update required packages

### PR DESCRIPTION
**Pull request by acrollet**
## Description

See #3303 

## Testing done

Automated testing.

## QA Instructions

Verify ACs:
- [ ] Update composer-patches to > 1.7.0
- [ ] Update oomphinc/composer-installers-extender to > 2.0.0
- [ ] use yaml-tasks instead of yaml-tests
- [ ] Update composer-drupal-optimizations to ^2
- [ ] drupal-console-extend-plugin is updated
- [ ] Unpin the composer version in BRD/DevOps repo or pin to version 2
- [ ] lando is updated to use composer 2
- [ ] tugboat is updated to use composer 2
- [ ] No non test-related composer errors saying some classes/dependencies won't autoload 
- [ ] a test content release works correctly on tugboat

### Select Team for PR review

- [x] `Core Application Team`
- [ ] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: .tugboat/config.yml [hunk 1]
```diff
@@ -78,8 +78,8 @@ services:
         - echo "apc.enable=1" >> /usr/local/etc/php/conf.d/php-cli.ini
         - echo "apc.enable_cli=1" >> /usr/local/etc/php/conf.d/php-cli.ini
 
-        # Downgrade to Composer 1 until https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3303 is done
-        - composer self-update --1
+        # Ensure that we're using version 2 of composer.
+        - composer self-update --2
```
Comment (ElijahLynn) [inline .tugboat/config.yml:82]: Nice, this is showing the diff now.

File: .lando.yml [hunk 2]
```diff
@@ -151,7 +151,7 @@ tooling:
   phpunit:
     service: appserver
     description: "Run VA PHPUnit tests found in va/tests/phpunit: lando phpunit. Syntax: lando phpunit"
-    cmd: "composer yaml-tasks va/tests/phpunit"
+    cmd: "composer yaml-tasks --tasks-file=tests.yml va/tests/phpunit"
```
[the remark this answers is not in the document]
Comment (acrollet) [reply .lando.yml:154]: whoops, inconsistent - I think composer should be fine.

**Comment by acrollet** [comment]
Verified the content release functionality:
Screen Shot 2021-03-10 at 10 33 34 AM
Screen Shot 2021-03-10 at 10 34 45 AM

**Comment by acrollet** [comment]
NB: it appears that composer 2 does not write the `patches_applied` sections that we are used to to composer.lock. I spot-checked several randomly chosen patches and confirmed that they were applied on the tugboat preview.

**Comment by ElijahLynn** [comment]
DevOps PR here > https://github.com/department-of-veterans-affairs/devops/pull/8669

**Comment by acrollet** [comment]
There was a 502 during `va/web/build`, re-running it and behat.

**Comment by acrollet** [comment]
> There was a 502 during `va/web/build`, re-running it and behat.

It was a blip, all tests passing.

**Comment by acrollet** [comment]
Ok, I think the 502 error is actually consistent. I'm seeing this in the test logs:

```
 ┃ | GetNodeLeadershipListingPages | 3s | 13 pages |                                                                    
 ┃                                                                                                                      
 ┃ Error: HTTP error: 502: Bad Gateway                                                                                  
 ┃ at Object.query (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/api.js:123:13)               ┃ at processTicksAndRejections (internal/process/task_queues.js:93:5)                                                  

 ┃ at runMicrotasks (<anonymous>)                                                                                       
 ┃ at async parallelQuery (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/api.js:225:22)       
 ┃ at async Promise.all (index 6)                                                                                       
 ┃ at async Object.getAllPagesViaIndividualGraphQlQueries                                                                ┃ (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/api.js:284:7)                               

 ┃ at async loadDrupal                                                                                                   ┃ at async getContentViaGraphQL                                                                                         ┃ (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/metalsmith-drupal.js:162:19)                 ┃ (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/metalsmith-drupal.js:231:7)                 

 ┃ Failed to pipe Drupal content into Metalsmith!                                                                       
 ┃                                                                                                                       ┃ at async Ware.<anonymous>                                                                                             ┃ (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/metalsmith-drupal.js:275:20)                
 ┃ /var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/index.js:285                                        
 ┃ if (err) throw err;                                                                                                  
 ┃ ^                                                                                                                    
 ┃ at async parallelQuery (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/api.js:225:22)       
 ┃ at async Promise.all (index 6)                                                                                       
 ┃                                                                                                                      
 ┃ Error: HTTP error: 502: Bad Gateway                                                                                  
 ┃ at Object.query (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/api.js:123:13)              
 ┃ at runMicrotasks (<anonymous>)                                                                                       
 ┃ at processTicksAndRejections (internal/process/task_queues.js:93:5)                                                  
 ┃ at async Object.getAllPagesViaIndividualGraphQlQueries                                                                ┃ (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/api.js:284:7)                               
 ┃ at async loadDrupal                                                                                                   ┃ (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/metalsmith-drupal.js:231:7)                 
 ┃ at async getContentViaGraphQL                                                                                         ┃ (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/metalsmith-drupal.js:162:19)                
 ┃                                                                                                                       ┃ at async Ware.<anonymous>                                                                                             ┃ (/var/lib/tugboat/docroot/vendor/va-gov/web/src/site/stages/build/drupal/metalsmith-drupal.js:275:20)                
 ┃ error Command failed with exit code 1.                                                                               
 ┃                                                                                                                      
```

And this in the Drupal logs from I believe around the same time:

Screen Shot 2021-03-11 at 8 11 04 AM

**Comment by acrollet** [comment]
Tests did not report back but all passed:

```
Executed all tasks
==================

 ------------------------ ------------------------------------------------------------------------------------------------------------------------------------------------------- ---------- 
  Task Results                                                                                                                                                                               
 ------------------------ ------------------------------------------------------------------------------------------------------------------------------------------------------- ---------- 
  va/tests/phpunit         phpunit --exclude-group disabled tests/phpunit --colors=always                                                                                          ✔ Passed  
  va/tests/phplint         composer va:test:lint                                                                                                                                   ✔ Passed  
  va/tests/accessibility   composer va:test:accessibility                                                                                                                          ✔ Passed  
  va/tests/behavioral      composer va:test:behavioral                                                                                                                             ✔ Passed  
  va/web/build             ./tests/scripts/build-web.sh                                                                                                                            ✔ Passed  
  va/tests/behat           cd tests/behat                                                                                                                                          ✔ Passed  
                           behat --colors                                                                                                                                                    
                                                                                                                                                                                             
  va/tests/revision-log    ./tests/scripts/check-revision-logs.sh                                                                                                                  ✔ Passed  
  va/tests/check-cer       ./tests/scripts/check-cer.sh                                                                                                                            ✔ Passed  
  va/tests/status-error    drush $DRUSH_ALIAS core-requirements --format=list --severity=2 | grep . && exit 1 || echo 'Success: No Drupal status requirement errors were found.'   ✔ Passed  
 ------------------------ ------------------------------------------------------------------------------------------------------------------------------------------------------- ---------- 
```

**Comment by ElijahLynn** [comment]
> Dashboard:
> 
> * https://tugboat.vfs.va.gov/604a66676a899460da3acbeb

This is a clone I made for testing phpunit failures:

```
 ┃ Time: 2.3 minutes, Memory: 615.00 MB                                                                                 
 ┃                                                                                                                      
 ┃ There were 2 failures:                                                                                               
 ┃                                                                                                                      
 ┃ 1) tests\phpunit\Migration\VaHealthCareLocalFacilityStatusMigrationTest::testVaLocalHealthCareFacilityStatusMigration ┃ with data set "Initial migration completes successfully" ('va_node_health
_care_local_fac...status',                   ┃ 'health_care_local_facility', '\n<!DOCTYPE html>\n<head><tit...tml>\n', array('vha_999999', 'normal', 'All Veteran    ┃ patients are ente...place.'),
 1, false)                                                                              
 ┃                                                                                                                      
 ┃ Failed asserting that 0 is identical to 1.                                                                           
 ┃                                                                                                                      
 ┃ /var/www/cms/tests/phpunit/Migration/VaHealthCareLocalFacilityStatusMigrationTest.php:53                             
 ┃                                                                                                                      
 ┃ 2) tests\phpunit\Migration\VaHealthCareLocalFacilityStatusMigrationTest::testVaLocalHealthCareFacilityStatusMigration ┃ with data set "Updated migration completes successfully" ('va_node_health
_care_local_fac...status',                   ┃ 'health_care_local_facility', '\n<!DOCTYPE html>\n<head><tit...tml>\n', array('vha_999999', 'notice', 'For the        ┃ continued health and ...demic
.'), 1, true)                                                                           
 ┃                                                                                                                      
 ┃ Failed asserting that 0 is identical to 1.                                                                           
 ┃                                                                                                                      
 ┃ /var/www/cms/tests/phpunit/Migration/VaHealthCareLocalFacilityStatusMigrationTest.php:53                             
 ┃                                                                                                                      
 ┃ FAILURES!                                                                                                            
 ┃ Tests: 80, Assertions: 246, Failures: 2.                                                                             
 ┃                                                                                                                      
 ✘ Process Failed in 02:18 
```

**Comment by acrollet** [comment]
Documenting that all tests passed after the most recent merge:

Screen Shot 2021-03-11 at 1 27 01 PM

**Pull request merged**